walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,84 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Set;
 
+/** 
+Algorithm:
+    1. Pick time ranges from events where attendees overlap with meeting request.
+    2. Now find the non overlapping time ranges which have duration more than request duration.
+*/
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    
+    private long meetingDuration;
+    private ArrayList<TimeRange> overlappingRanges;
+    private Collection<String> requestedAttendees;
+    private Collection<TimeRange> requiredRanges;
```

Comment:
this makes your class stateful. Please have these as part of the method.
Reason, think of a scenario where multiple request come to this class, these variable will already have a pre-existing value which might lead to a non deterministic outcome.

Reply:
Done.